pkg/pillar/hypervisor/xen.go

```diff
@@ -198,7 +198,7 @@ func (ctx xenContext) CreateDomConfig(domainName string, config types.DomainConf
 		file.WriteString(fmt.Sprintf("bootloader = \"%s\"\n",
 			bootLoader))
 	}
-	if config.EnableVnc {
+	if config.EnableVnc || config.EnableVncShimVm {
```

Comment:
I am not sure that it is expected from the user that if `EnableVncShimVm` is set, that also normal VNC is enabled.

Reply:
That is a good suspect - either we enable vnc -> shim vnc in this order, either we enable vnc and shim vnc separately.

Reply:
thank you @christoph-zededa , I'd rather have it cascading like
EnableVnc                      [ ]
|_ EnableVncShimVM    [ ]
so EnableVncShimVM cannot be set if EnableVnc is not set. The controller should be implemented accordingly - I'll mention this in the documentation.

Reply:
Perhaps it is also possible to check it in eve and send back an error to the controller?

Reply:
Yes, did it.